README.md
# EPC-Programmer

changelog.txt
11/15/18 - V1.0 release

	- no user input
	- basic functionality to program generic EPC

11/17/18 - V1.1 release

	- added initial modular file structuring
	- preparing GUI framework for next major release


EPC.py
from tkinter import *

root = Tk()
windowSize = "960x540"
epc_mods = ["Passwords", "Network Settings", "Outlet Names"]
epc_defaults = ["1234", "CAL1108cal", "192.168.168.206", "192.168.168.168", "8.8.8.8"]
outl_defaults = ["DLI Controller", "Outlet 1", "Outlet 2", "Outlet 3", "Outlet 4", "Outlet 5", "Outlet 6", "Outlet 7", "Outlet 8"]

def epc_main():
    import refers
    epc = refers.funcss
    epc.login(Entry.get(e_0))
    epc.set_outlet()
    epc.set_delay()
    #epc.set_units(Entry.get(e_5),Entry.get(e_6),Entry.get(e_7),Entry.get(e_8),Entry.get(e_9),Entry.get(e_10),Entry.get(e_11),Entry.get(e_12),Entry.get(e_13))
    epc.set_powerloss()
    epc.set_wifi()
    epc.set_plaintext()
    epc.set_access()
    epc.set_admincred(Entry.get(e_0),Entry.get(e_1))
    epc.login(Entry.get(e_1))
    epc.set_outlet()
    epc.set_lan(Entry.get(e_2),Entry.get(e_3),Entry.get(e_4))
    return

root.title("Valkyrie EPC Programmer")
root.geometry(windowSize)

bt_submit = Button(root, text="Submit", command=epc_main)
bt_submit.place(x=550, y=360, width=120, height=25)


Label(text=epc_mods[0], relief=RIDGE,width=40).place(x=480,y=0)

Label(text="Initial Password:").place(x=480,y=40)
v = StringVar(root, value=epc_defaults[0])
e_0 = Entry(textvariable=v)
e_0.place(x=580,y=40)
Label(text="Final Password:").place(x=480,y=80)
v = StringVar(root, value=epc_defaults[1])
e_1 = Entry(textvariable=v)
e_1.place(x=580,y=80)


Label(text=epc_mods[1], relief=RIDGE,width=40).place(x=480,y=120)

Label(text="IP Address:").place(x=480,y=160)
v = StringVar(root, value=epc_defaults[2])
e_2 = Entry(textvariable=v)
e_2.place(x=580,y=160)
Label(text="Gateway:").place(x=480,y=200)
v = StringVar(root, value=epc_defaults[3])
e_3 = Entry(textvariable=v)
e_3.place(x=580,y=200)
Label(text="DNS Server:").place(x=480,y=240)
v = StringVar(root, value=epc_defaults[4])
e_4 = Entry(textvariable=v)
e_4.place(x=580,y=240)


Label(text=epc_mods[2], relief=RIDGE,width=40).place(x=0)

Label(text="Controller Name:").place(x=0,y=40)
v = StringVar(root, value=outl_defaults[0])
e_5 = Entry(textvariable=v)
e_5.place(x=100,y=40)
Label(text="Outlet 1 Name:").place(x=0,y=80)
v = StringVar(root, value=outl_defaults[1])
e_6 = Entry(textvariable=v)
e_6.place(x=100,y=80)
Label(text="Outlet 2 Name:").place(x=0,y=120)
v = StringVar(root, value=outl_defaults[2])
e_7 = Entry(textvariable=v)
e_7.place(x=100,y=120)
Label(text="Outlet 3 Name:").place(x=0,y=160)
v = StringVar(root, value=outl_defaults[3])
e_8 = Entry(textvariable=v)
e_8.place(x=100,y=160)
Label(text="Outlet 4 Name:").place(x=0,y=200)
v = StringVar(root, value=outl_defaults[4])
e_9 = Entry(textvariable=v)
e_9.place(x=100,y=200)
Label(text="Outlet 5 Name:").place(x=0,y=240)
v = StringVar(root, value=outl_defaults[5])
e_10 = Entry(textvariable=v)
e_10.place(x=100,y=240)
Label(text="Outlet 6 Name:").place(x=0,y=280)
v = StringVar(root, value=outl_defaults[6])
e_11 = Entry(textvariable=v)
e_11.place(x=100,y=280)
Label(text="Outlet 7 Name:").place(x=0,y=320)
v = StringVar(root, value=outl_defaults[7])
e_12 = Entry(textvariable=v)
e_12.place(x=100,y=320)
Label(text="Outlet 8 Name:").place(x=0,y=360)
v = StringVar(root, value=outl_defaults[8])
e_13 = Entry(textvariable=v)
e_13.place(x=100,y=360)

root.mainloop()


EPC 1.0.py
import time
from selenium import webdriver

options = webdriver.ChromeOptions()
options.add_argument('--ignore-certificate-errors')
options.add_argument("--test-type")
options.binary_location = "/usr/bin/chromium"
driver = webdriver.Chrome()
driver.get("http://192.168.0.100")

LoginVals = ["//*[@id='login']/table/tbody/tr[1]/td[2]/input", "admin", "//*[@id='login']/table/tbody/tr[2]/td[2]/input", "//*[@id='login']/table/tbody/tr[3]/td[2]/input[1]"]
OCvals = ["/html/body/table/tbody/tr/td[2]/table[3]/tbody/tr[3]/td/a", "/html/body/table/tbody/tr/td[1]/table/tbody/tr[1]/td/table/tbody/tr[3]/td/a"]
DelayVals = ["/html/body/table/tbody/tr/td[2]/form[2]/div/table/tbody/tr[1]/td[2]/input", "/html/body/table/tbody/tr/td[2]/form[2]/div/table/tbody/tr[3]/td[2]/input", "/html/body/table/tbody/tr/td[2]/form[2]/div/table/tbody/tr[7]/td/input"]
PowerlossVals = ["/html/body/table/tbody/tr/td[2]/form[3]/div/table/tbody/tr[1]/td[2]/input[2]", "/html/body/table/tbody/tr/td[2]/form[3]/div/table/tbody/tr[3]/td/input"]
WifiVals = ["/html/body/table/tbody/tr/td[2]/form[7]/div[2]/table/tbody/tr[1]/td[2]/input", "/html/body/table/tbody/tr/td[2]/form[7]/div[2]/table/tbody/tr[13]/td/input", "/html/body/table/tbody/tr/td/p[3]/a"]
PlaintextVals = ["/html/body/table/tbody/tr/td[2]/form[12]/div/table/tbody/tr[1]/td[2]/input", "/html/body/table/tbody/tr/td[2]/form[12]/div/table/tbody/tr[6]/td/input"]
AccessnameVals = ["//*[@id='login1']", "administrator", "//*[@id='password1']", "CAL1108cal"]
AccessOutletVals = ["//*[@id='perm1_1']", "//*[@id='perm1_2']", "//*[@id='perm1_3']", "//*[@id='perm1_4']", "//*[@id='perm1_5']", "//*[@id='perm1_6']", "//*[@id='perm1_7']", "//*[@id='perm1_8']", "/html/body/table/tbody/tr/td[2]/form[11]/div[2]/table/tbody/tr[3]/td[11]/input"]
AdminVals = ["/html/body/table/tbody/tr/td[2]/form[9]/div[2]/table/tbody/tr[2]/td[2]/input", "/html/body/table/tbody/tr/td[2]/form[9]/div[2]/table/tbody/tr[3]/td[2]/input", "/html/body/table/tbody/tr/td[2]/form[9]/div[2]/table/tbody/tr[4]/td[2]/input", "/html/body/table/tbody/tr/td[2]/form[9]/div[2]/table/tbody/tr[5]/td[1]/input"]
LANxvals = ["/html/body/table/tbody/tr/td[2]/form[6]/div/table/tbody/tr[3]/td[2]/input", "/html/body/table/tbody/tr/td[2]/form[6]/div/table/tbody/tr[5]/td[2]/input", "/html/body/table/tbody/tr/td[2]/form[6]/div/table/tbody/tr[6]/td[2]/input", "/html/body/table/tbody/tr/td[2]/form[6]/div/table/tbody/tr[7]/td/input", "/html/body/table/tbody/tr/td/p[3]/a"]
LANvals = ["192.168.168.206", "192.168.168.168", "8.8.8.8"]


def clear(x):
    driver.find_element_by_xpath(x).clear() #clears the default start time
    time.sleep(0.001)
    return
def login(y):
    driver.find_element_by_xpath(LoginVals[0]).send_keys(LoginVals[1])
    driver.find_element_by_xpath(LoginVals[2]).send_keys(y)
    driver.find_element_by_xpath(LoginVals[3]).click()
    time.sleep(0.5)
    return
def set_outlet():
    driver.find_element_by_xpath(OCvals[0]).click()
    driver.find_element_by_xpath(OCvals[1]).click()
    return
def set_delay():
    clear(DelayVals[0])
    driver.find_element_by_xpath(DelayVals[0]).send_keys("3")
    clear(DelayVals[1])
    driver.find_element_by_xpath(DelayVals[1]).send_keys("3")
    driver.find_element_by_xpath(DelayVals[2]).click()
    time.sleep(1)
    return
def set_powerloss():
    driver.find_element_by_xpath(PowerlossVals[0]).click()
    driver.find_element_by_xpath(PowerlossVals[1]).click()
    time.sleep(1)
    return
def set_wifi():
    driver.find_element_by_xpath(WifiVals[0]).click()
    driver.find_element_by_xpath(WifiVals[1]).click()
    driver.switch_to_alert().accept()
    time.sleep(1)
    driver.find_element_by_xpath(WifiVals[2]).click()
    time.sleep(0.5)
    return
def set_plaintext():
    driver.find_element_by_xpath(PlaintextVals[0]).click()
    driver.find_element_by_xpath(PlaintextVals[1]).click()
    time.sleep(0.5)
def set_access():
    driver.find_element_by_xpath(AccessnameVals[0]).send_keys(AccessnameVals[1])
    driver.find_element_by_xpath(AccessnameVals[2]).send_keys(AccessnameVals[3])
    time.sleep(0.1)
    driver.find_element_by_xpath(AccessOutletVals[0]).click()
    driver.find_element_by_xpath(AccessOutletVals[1]).click()
    driver.find_element_by_xpath(AccessOutletVals[2]).click()
    driver.find_element_by_xpath(AccessOutletVals[3]).click()
    driver.find_element_by_xpath(AccessOutletVals[4]).click()
    driver.find_element_by_xpath(AccessOutletVals[5]).click()
    driver.find_element_by_xpath(AccessOutletVals[6]).click()
    driver.find_element_by_xpath(AccessOutletVals[7]).click()
    driver.find_element_by_xpath(AccessOutletVals[8]).click()
def set_admincred():
    time.sleep(0.5)
    driver.find_element_by_xpath(AdminVals[0]).send_keys("1234")
    driver.find_element_by_xpath(AdminVals[1]).send_keys("CAL1108cal")
    driver.find_element_by_xpath(AdminVals[2]).send_keys("CAL1108cal")
    driver.find_element_by_xpath(AdminVals[3]).click()
    time.sleep(0.5)
    return
def set_lan():
    clear(LANxvals[0])
    driver.find_element_by_xpath(LANxvals[0]).send_keys(LANvals[0])
    driver.find_element_by_xpath(LANxvals[1]).send_keys(LANvals[1])
    driver.find_element_by_xpath(LANxvals[2]).send_keys(LANvals[2])
    driver.find_element_by_xpath(LANxvals[3]).click()
    driver.switch_to_alert().accept()
    time.sleep(1)
    driver.switch_to_default_content()
    driver.find_element_by_xpath("/html/body/table/tbody/tr/td/p[3]/a").click()
    return
def main_EPC():
    login("1234")
    set_outlet()
    set_delay()
    set_powerloss()
    set_wifi()
    set_plaintext()
    set_access()
    set_admincred()
    login("CAL1108cal")
    set_outlet()
    set_lan()
    return

main_EPC()


refers.py
import time
from selenium import webdriver

LoginVals = ["//*[@id='login']/table/tbody/tr[1]/td[2]/input", "admin", "//*[@id='login']/table/tbody/tr[2]/td[2]/input", "//*[@id='login']/table/tbody/tr[3]/td[2]/input[1]"]
OCvals = ["/html/body/table/tbody/tr/td[2]/table[3]/tbody/tr[3]/td/a", "/html/body/table/tbody/tr/td[1]/table/tbody/tr[1]/td/table/tbody/tr[3]/td/a"]
DelayVals = ["/html/body/table/tbody/tr/td[2]/form[2]/div/table/tbody/tr[1]/td[2]/input", "/html/body/table/tbody/tr/td[2]/form[2]/div/table/tbody/tr[3]/td[2]/input", "/html/body/table/tbody/tr/td[2]/form[2]/div/table/tbody/tr[7]/td/input"]
PowerlossVals = ["/html/body/table/tbody/tr/td[2]/form[3]/div/table/tbody/tr[1]/td[2]/input[2]", "/html/body/table/tbody/tr/td[2]/form[3]/div/table/tbody/tr[3]/td/input"]
WifiVals = ["/html/body/table/tbody/tr/td[2]/form[7]/div[2]/table/tbody/tr[1]/td[2]/input", "/html/body/table/tbody/tr/td[2]/form[7]/div[2]/table/tbody/tr[13]/td/input", "/html/body/table/tbody/tr/td/p[3]/a"]
PlaintextVals = ["/html/body/table/tbody/tr/td[2]/form[12]/div/table/tbody/tr[1]/td[2]/input", "/html/body/table/tbody/tr/td[2]/form[12]/div/table/tbody/tr[6]/td/input"]
AccessnameVals = ["//*[@id='login1']", "administrator", "//*[@id='password1']", "CAL1108cal"]
AccessOutletVals = ["//*[@id='perm1_1']", "//*[@id='perm1_2']", "//*[@id='perm1_3']", "//*[@id='perm1_4']", "//*[@id='perm1_5']", "//*[@id='perm1_6']", "//*[@id='perm1_7']", "//*[@id='perm1_8']", "/html/body/table/tbody/tr/td[2]/form[11]/div[2]/table/tbody/tr[3]/td[11]/input"]
AdminVals = ["/html/body/table/tbody/tr/td[2]/form[9]/div[2]/table/tbody/tr[2]/td[2]/input", "/html/body/table/tbody/tr/td[2]/form[9]/div[2]/table/tbody/tr[3]/td[2]/input", "/html/body/table/tbody/tr/td[2]/form[9]/div[2]/table/tbody/tr[4]/td[2]/input", "/html/body/table/tbody/tr/td[2]/form[9]/div[2]/table/tbody/tr[5]/td[1]/input"]
LANxvals = ["/html/body/table/tbody/tr/td[2]/form[6]/div/table/tbody/tr[3]/td[2]/input", "/html/body/table/tbody/tr/td[2]/form[6]/div/table/tbody/tr[5]/td[2]/input", "/html/body/table/tbody/tr/td[2]/form[6]/div/table/tbody/tr[6]/td[2]/input", "/html/body/table/tbody/tr/td[2]/form[6]/div/table/tbody/tr[7]/td/input", "/html/body/table/tbody/tr/td/p[3]/a"]
LANvals = ["192.168.168.206", "192.168.168.168", "8.8.8.8"]
Unitxvals = ["/html/body/table/tbody/tr/td[2]/form[1]/div/table/tbody/tr[1]/td[2]/input","/html/body/table/tbody/tr/td[2]/form[1]/div/table/tbody/tr[2]/td[2]/input", "/html/body/table/tbody/tr/td[2]/form[1]/div/table/tbody/tr[3]/td[2]/input", "/html/body/table/tbody/tr/td[2]/form[1]/div/table/tbody/tr[4]/td[2]/input", "/html/body/table/tbody/tr/td[2]/form[1]/div/table/tbody/tr[5]/td[2]/input", "/html/body/table/tbody/tr/td[2]/form[1]/div/table/tbody/tr[6]/td[2]/input", "/html/body/table/tbody/tr/td[2]/form[1]/div/table/tbody/tr[7]/td[2]/input", "/html/body/table/tbody/tr/td[2]/form[1]/div/table/tbody/tr[8]/td[2]/input", "/html/body/table/tbody/tr/td[2]/form[1]/div/table/tbody/tr[9]/td[2]/input"]

options = webdriver.ChromeOptions()
options.add_argument('--ignore-certificate-errors')
options.add_argument("--test-type")
options.binary_location = "/usr/bin/chromium"
driver = webdriver.Chrome()
driver.get("http://192.168.0.100")

def clear(x):
    driver.find_element_by_xpath(x).clear() #clears the default start time
    time.sleep(0.001)
    return

class funcss:
    def login(y):
        driver.find_element_by_xpath(LoginVals[0]).send_keys(LoginVals[1])
        driver.find_element_by_xpath(LoginVals[2]).send_keys(y)
        driver.find_element_by_xpath(LoginVals[3]).click()
        time.sleep(0.5)
        return
    def set_outlet():
        driver.find_element_by_xpath(OCvals[0]).click()
        driver.find_element_by_xpath(OCvals[1]).click()
        return
    def set_delay():
        clear(DelayVals[0])
        driver.find_element_by_xpath(DelayVals[0]).send_keys("3")
        clear(DelayVals[1])
        driver.find_element_by_xpath(DelayVals[1]).send_keys("3")
        time.sleep(1)
        driver.find_element_by_xpath(DelayVals[2]).click()
        time.sleep(1)
        return
    def set_powerloss():
        driver.find_element_by_xpath(PowerlossVals[0]).click()
        driver.find_element_by_xpath(PowerlossVals[1]).click()
        time.sleep(1)
        return
    def set_wifi():
        driver.find_element_by_xpath(WifiVals[0]).click()
        driver.find_element_by_xpath(WifiVals[1]).click()
        driver.switch_to_alert().accept()
        time.sleep(1)
        driver.find_element_by_xpath(WifiVals[2]).click()
        time.sleep(0.5)
        return
    def set_plaintext():
        driver.find_element_by_xpath(PlaintextVals[0]).click()
        driver.find_element_by_xpath(PlaintextVals[1]).click()
        time.sleep(0.5)
    def set_access():
        clear(AccessnameVals[0])
        driver.find_element_by_xpath(AccessnameVals[0]).send_keys(AccessnameVals[1])
        clear(AccessnameVals[2])
        driver.find_element_by_xpath(AccessnameVals[2]).send_keys(AccessnameVals[3])
        time.sleep(0.1)
        driver.find_element_by_xpath(AccessOutletVals[0]).click()
        driver.find_element_by_xpath(AccessOutletVals[1]).click()
        driver.find_element_by_xpath(AccessOutletVals[2]).click()
        driver.find_element_by_xpath(AccessOutletVals[3]).click()
        driver.find_element_by_xpath(AccessOutletVals[4]).click()
        driver.find_element_by_xpath(AccessOutletVals[5]).click()
        driver.find_element_by_xpath(AccessOutletVals[6]).click()
        driver.find_element_by_xpath(AccessOutletVals[7]).click()
        driver.find_element_by_xpath(AccessOutletVals[8]).click()
    def set_admincred(t,u):
        time.sleep(0.5)
        driver.find_element_by_xpath(AdminVals[0]).send_keys(u)
        driver.find_element_by_xpath(AdminVals[1]).send_keys(t)
        driver.find_element_by_xpath(AdminVals[2]).send_keys(t)
        driver.find_element_by_xpath(AdminVals[3]).click()
        time.sleep(0.5)
        return
    def set_lan(q,w,e):
        clear(LANxvals[0])
        driver.find_element_by_xpath(LANxvals[0]).send_keys(q)
        driver.find_element_by_xpath(LANxvals[1]).send_keys(w)
        driver.find_element_by_xpath(LANxvals[2]).send_keys(e)
        driver.find_element_by_xpath(LANxvals[3]).click()
        driver.switch_to_alert().accept()
        time.sleep(1)
        driver.switch_to_default_content()
        driver.find_element_by_xpath("/html/body/table/tbody/tr/td/p[3]/a").click()
        return
    def set_units(a,b,c,d,e,f,g,h,i):
        time.sleep(1)
        clear(Unitxvals[0])
        driver.find_elements_by_xpath(Unitxvals[0]).send_keys(a)
        clear(Unitxvals[1])
        driver.find_elements_by_xpath(Unitxvals[1]).send_keys("testt")
        clear(Unitxvals[2])
        driver.find_elements_by_xpath(Unitxvals[2]).send_keys(c)
        clear(Unitxvals[3])
        driver.find_elements_by_xpath(Unitxvals[3]).send_keys(d)
        clear(Unitxvals[4])
        driver.find_elements_by_xpath(Unitxvals[4]).send_keys(e)
        clear(Unitxvals[5])
        driver.find_elements_by_xpath(Unitxvals[5]).send_keys(f)
        clear(Unitxvals[6])
        driver.find_elements_by_xpath(Unitxvals[6]).send_keys(g)
        clear(Unitxvals[7])
        driver.find_elements_by_xpath(Unitxvals[7]).send_keys(h)
        clear(Unitxvals[8])
        driver.find_elements_by_xpath(Unitxvals[8]).send_keys(i)
        return

funcss.login("CAL1108cal")
funcss.set_outlet()
funcss.set_delay()
